nucleus/cluster/admin/src/main/java/fish/payara/admin/cluster/ListDeploymentGroupsCommand.java

```diff
@@ -0,0 +1,105 @@
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.admin.cluster;
+
+import com.sun.enterprise.config.serverbeans.Domain;
+import fish.payara.enterprise.config.serverbeans.DeploymentGroup;
+import fish.payara.enterprise.config.serverbeans.DeploymentGroups;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import javax.inject.Inject;
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.I18n;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.CommandLock;
+import org.glassfish.api.admin.RestEndpoint;
+import org.glassfish.api.admin.RestEndpoints;
+import org.glassfish.hk2.api.PerLookup;
+import org.jvnet.hk2.annotations.Service;
+
+/**
+ * Command to list Deployment Groups
+ *
+ * @since 5.0
+ * @author Susan Rai
+ */
+@Service(name = "list-deployment-group")
+@I18n("list.deployment.group.command")
+@PerLookup
+@CommandLock(CommandLock.LockType.NONE)
+@RestEndpoints({
+    @RestEndpoint(configBean = DeploymentGroups.class,
+            opType = RestEndpoint.OpType.GET,
+            path = "list-deployment-group",
+            description = "Command to list Deployment Groups")
+})
+public class ListDeploymentGroupsCommand implements AdminCommand {
+
+    @Inject
+    Domain domain;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        final ActionReport report = context.getActionReport();
+
+        DeploymentGroups deploymentGroups = domain.getDeploymentGroups();
+        List<DeploymentGroup> listOfDeploymentGroup = deploymentGroups.getDeploymentGroup();
+
+        if (listOfDeploymentGroup.isEmpty()) {
+            report.appendMessage("No Deployment Group has been created");
+        } else {
+            StringBuffer sb = new StringBuffer();
+            sb.append("List of Deployment Groups" + ":\n");
+            Properties extrasProps = new Properties();
+            ArrayList<String> deploymentGroupNames = new ArrayList<String>();
+            for (DeploymentGroup deploymentGroup : listOfDeploymentGroup) {
+                sb.append("\t" + deploymentGroup.getName() + "\n");
+                deploymentGroupNames.add(deploymentGroup.getName());
+            }
+            extrasProps.put("listOfDeploymentGroups", deploymentGroupNames);
+            report.setMessage(sb.toString());
+            report.setExtraProperties(extrasProps);
+            report.setActionExitCode(ActionReport.ExitCode.SUCCESS);
```

Comment:
This is unnecessary - you only need to set it to `SUCCESS` if it's been set to `FAILURE` somewhere else.